test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelpersTest.cs

```diff
@@ -148,5 +148,110 @@ public class TagHelpersTests
             var responseContent = await response.Content.ReadAsStringAsync();
             Assert.Equal(expectedContent, responseContent);
         }
+
+        [Fact]
+        public async Task CacheTagHelper_CanCachePortionsOfViewsPartialViewsAndViewComponents()
```

Comment:
I'll get that done prior to checking this in. Right now this is based off a separate branch since the Tag helper changes break the existing ones and I had to comment them out to get this one to work. Rebasing on #1770 would require redoing this PR since GitHub does not track changes to the base branch.